libraries/ui-library/src/components/six-select/six-select.tsx

```diff
@@ -493,7 +479,7 @@ export class SixSelect {
       item.checked = value.includes(item.value);
     });
 
-    const checkedItems = getCheckedItems(this.getValueAsArray(), mainItems);
+    const checkedItems = getCheckedItems(Array.isArray(this.value) ? this.value : [this.value], mainItems);
```

Comment:
Might be worth to add another utility function to replace the two occurences of this expression:
```ts
export function convertToValidArrayValue(value: unknown): string[] {
  return convertToValidValue(value, true) as string[];
}
```